src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -0,0 +1,491 @@
+package strawman
+package collection.mutable
+
+import scala._
+import scala.collection.{generic, mutable}
```

Comment:
We don’t expect massive public API breakage, but internal API is very different because it has been designed from the beginning with non strict collections in mind (which is not the case of the old collections).

But you made a point: we have no design docs. I’ll fix that.